core/controllers/skill_editor.py

```diff
@@ -286,10 +286,10 @@ class SkillDataHandler(base.BaseHandler):
         'comma_separated_skill_ids': {
             'schema': {
                 'type': 'basestring',
-                'validators': [{
-                    'id': 'is_regex_matched',
-                    'regex_pattern': constants.ENTITY_ID_REGEX
-                }]
+                # 'validators': [{
+                #     'id': 'is_regex_matched',
+                #     'regex_pattern': constants.ENTITY_ID_REGEX
+                # }]
```

Comment:
I wasn't sure if this validator was necessary because the skill IDs are extracted from the list and validated later in the get() method of this class. This validator also failed when comma-separated skill IDs were entered because a comma is not allowed by the regex pattern.

Reply:
Shouldn't this match a regex for having comma separated entity ids?

Reply:
I don't believe I saw a regex pattern for that. Is it ok if I add one?

Reply:
Use `JsonEncodedInString` for this and modify the frontend code for this. You can look in the other usages of `JsonEncodedInString` in our codebase.

Reply:
I looked at other handlers like ConceptCardDataHandler that use JsonEncodedInString as well and tried to model the SkillDataHandler's code similarly (i.e. using JSON.stringify in the frontend). Unfortunately it didn't work and I am having a difficult time trying to debug. You can see my code in [this commit](https://github.com/oppia/oppia/pull/15306/commits/029ceeafa1256b35946b9d8200c46eb66c3426ff) which I later reverted in order to push. Could you clarify your request a bit?

Reply:
Why it didn't work and what was the problem with debugging?